src/main/java/com/weaponcharges/WeaponChargesItemOverlay.java

```diff
@@ -119,6 +126,39 @@ public void renderItemOverlay(Graphics2D graphics, int itemId, WidgetItem itemWi
 			found = true;
 		}
 
+		if (ChargedWeapon.SERPENTINE_HELM.getItemIds().contains(itemId)) {
+			DisplayWhen displayWhen = DisplayWhenNoDefault.getDisplayWhen(config.serpentine_helm_Display(), config.defaultDisplay());
+			Integer charges = null;
+			ChargedWeapon chargedWeapon = ChargedWeapon.SERPENTINE_HELM;
+			charges = plugin.getCharges(chargedWeapon);
+			if (charges == null)
+			{
+				topText.setText("??.?%");
+			} else {
+				if (displayWhen == DisplayWhen.NEVER && !plugin.isShowChargesKeyIsDown()) return;
+				NumberFormat df = new DecimalFormat("##0.0");
+				float scalesLeftPercent = (float) charges / chargedWeapon.rechargeAmount;
+				df.setRoundingMode(RoundingMode.DOWN);
+				if (config.serpentine_helm_DisplayStyle() == SerpModes.SCALES) {
+					topText.setText(charges.toString());
+				} else if (config.serpentine_helm_DisplayStyle() == SerpModes.PERCENT) {
+					topText.setText(df.format((scalesLeftPercent * 100)) + "%");
+					if (df.format((scalesLeftPercent * 100)).equals(df.format(0.0)) && charges > 0) {
+						topText.setText("1.0%");
```

Comment:
should be 0.1%.

Reply:
or whatever the smallest non-0-looking value is.

Reply:
actually, is a non-0 value even capable of producing 0.0%? have you tested that?

Reply:
oh I see, that's how I do my blowpipe code. nevermind.

So make sure that when you remove the duplicated code you remove it from the blowpipe code also.

Reply:
> So make sure that when you remove the **duplicated code** you remove it from the blowpipe code also.

By "duplicated code" are you referring to the entire conditional check there?
```java
if (df.format((scalesLeftPercent * 100)).equals(df.format(0.0)) && charges > 0) {
	topText.setText("1.0%");
}
```
I can remove this from Serpentine and Blowpipe.

Reply:
De-duplicated; leaving in until it's confirmed to not be needed. It is a nice edge-case to have covered.